orangecontrib/network/widgets/OWNxTwitterGraph.py

```diff
@@ -0,0 +1,248 @@
+
+            self.submit_button = gui.button(self.controlArea, self, "OK", self.accept)
+
+            self.load_credentials()
+
+        def load_credentials(self):
+            self.key_edit.setText(self.cm_key.key)
+            self.secret_edit.setText(self.cm_secret.key)
+
+        def save_credentials(self):
+            self.cm_key.key = self.key_input
+            self.cm_secret.key = self.secret_input
+
+        def check_credentials(self):
+            c = txt_twitter.Credentials(self.key_input, self.secret_input)
+            if self.credentials != c:
+                if c.valid:
+                    self.save_credentials()
+                else:
+                    c = None
+                self.credentials = c
+
+        def accept(self, silent=False):
+            if not silent: self.Error.invalid_credentials.clear()
+            self.check_credentials()
+            if self.credentials and self.credentials.valid:
+                super().accept()
+            elif not silent:
+                self.Error.invalid_credentials()
+
+    name = "Twitter User Graph"
+    description = "Create a graph of Twitter users."
+    icon = "icons/Twitter.svg"
+    priority = 6470
+    outputs = [("Followers", network.Graph),
+               ("Following", network.Graph),
+               ("All", network.Graph)]
+
+    want_main_area = False
+
+
+    def __init__(self):
+        super().__init__()
+
+        self.n_all = 0
+        self.n_followers = 0
+        self.n_following = 0
+        self.on_rate_limit = None
+        self.api_dlg = self.APICredentialsDialog(self)
+
+        # GUI
+        # Set API key button.
+        key_dialog_button = gui.button(self.controlArea, self, 'Twitter API Key',
+                                       callback=self.open_key_dialog,
+                                       tooltip="Set the API key for this widget.")
+        key_dialog_button.setFocusPolicy(QtCore.Qt.NoFocus)
+        box = gui.widgetBox(self.controlArea, "Info")
+        box.layout().addWidget(QtGui.QLabel("Users:"))
+        self.users = QtGui.QTextEdit()
+        box.layout().addWidget(self.users)
+        gui.label(box, self, 'Following: %(n_following)d\n'
+                             'Followers: %(n_followers)d\n'
+                             'All: %(n_all)d')
+        self.button = gui.button(box, self, "Create graph", self.fetch_users)
+
+    def open_key_dialog(self):
+        self.api_dlg.exec_()
+
+    def fetch_users(self):
+        CONSUMER_KEY = CredentialManager('Twitter API Key').key
+        CONSUMER_SECRET = CredentialManager('Twitter API Secret').key
+        OAUTH_TOKEN = ''
+        OAUTH_TOKEN_SECRET = ''
+        auth = twitter.oauth.OAuth(OAUTH_TOKEN, OAUTH_TOKEN_SECRET, CONSUMER_KEY, CONSUMER_SECRET)
+        t = twitter.Twitter(auth=auth)
+
+        followers_graph = nx.Graph()
+        following_graph = nx.Graph()
+        all_users = nx.Graph()
+        users = self.users.toPlainText().split("\n")
```

Comment:
Must be something in the way you specify users. This just splits on characters not in (`A-Z`, `a-z`, `0-9`, and `_`), might need adapting (to accept `@` or skip empty strings or something like that). Point was: it's not necessarily complicated. :smile: